infrastructure/async/src/main/java/tech/pegasys/teku/infrastructure/async/LimitedTaskQueue.java

```diff
@@ -61,21 +62,24 @@ private LimitedTaskQueue(final TaskQueue delegate, final int maximumQueueSize) {
   }
 
   @Override
-  public synchronized <T> SafeFuture<T> queueTask(final Supplier<SafeFuture<T>> request) {
-    if (delegate.getQueuedTasksCount() >= maximumQueueSize) {
-      rejectedTaskCount++;
-      return SafeFuture.failedFuture(new QueueIsFullException());
+  public <T> SafeFuture<T> queueTask(final Supplier<SafeFuture<T>> request) {
+    synchronized (delegate) {
```

Comment:
I think we could drop even this synchronization. It's high pressure cutoff for RPC, we don't care about eventual consistency, two clients will not get their data now instead of one, not a big deal